service/api_logic/auth_strategy.py

```diff
@@ -0,0 +1,96 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+import bcrypt
+import requests
+from flask import current_app, request
+from oauthlib.oauth2 import WebApplicationClient
+from database.models import User
+from dto.api_input import InputUserLogInDTO
+from dto.api_output import OutputLogin
+from exept.exeptions import IncorrectUserDataError, UserDoesNotExistError, IncorrectLogInStrategyMethod, \
+    InvalidDataForGoogleLogIn
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+class AuthMethods(Enum):
```

Comment:
Seems to me like something that can be used in many projects of our program (API layer and BusinessLayer at least)

I suggest moving this to some separate place. You can create a separate "project" which would hold all such "global" models